spec/models/user_spec.rb

```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+  subject { users(:bob_user) }
+
+  it { should define_enum_for(:favorite_cause).backed_by_column_of_type(:string) }
+
+  it { should have_many(:user_favorite_organizations).dependent(:destroy) }
+  it { should have_many(:donations).dependent(:destroy) }
+  it do
+    should have_many(:favorite_organizations)
+      .through(:user_favorite_organizations)
+      .source(:organization)
```

Comment:
I had no idea it was possible to be this specific with the shoulda matchers. This is good 👍